File: src/main/java/org/taktik/icure/logic/impl/filter/code/CodeByRegionTypeLabelLanguageFilter.java
package org.taktik.icure.logic.impl.filter.code;

import org.springframework.beans.factory.annotation.Autowired;
import org.taktik.icure.entities.base.Code;
import org.taktik.icure.logic.CodeLogic;
import org.taktik.icure.logic.impl.filter.Filters;

import java.util.HashSet;
import java.util.Set;

public class CodeByRegionTypeLabelLanguageFilter implements org.taktik.icure.logic.impl.filter.Filter<String, Code, org.taktik.icure.dto.filter.code.CodeByRegionTypeLabelLanguageFilter> {
	CodeLogic codeLogic;

	@Override
	public Set<String> resolve(org.taktik.icure.dto.filter.code.CodeByRegionTypeLabelLanguageFilter filter, Filters context) {
		return new HashSet<>(codeLogic.listCodeIdsByLabel(filter.getRegion(), filter.getLanguage(), filter.getType(), filter.getLabel()));
	}

	@Autowired
	public void setCodeLogic(CodeLogic codeLogic) {
		this.codeLogic = codeLogic;
	}
}

File: src/main/kotlin/org/taktik/icure/asynclogic/impl/filter/code/CodeByRegionTypeLabelLanguageFilter.kt
package org.taktik.icure.asynclogic.impl.filter.code

import kotlinx.coroutines.flow.Flow
import org.taktik.icure.asynclogic.CodeLogic
import org.taktik.icure.asynclogic.impl.filter.Filter
import org.taktik.icure.asynclogic.impl.filter.Filters
import org.taktik.icure.domain.filter.code.CodeByRegionTypeLabelLanguageFilter
import org.taktik.icure.entities.base.Code

class CodeByRegionTypeLabelLanguageFilter(private val codeLogic: CodeLogic) : Filter<String, Code, CodeByRegionTypeLabelLanguageFilter> {
    override fun resolve(filter: CodeByRegionTypeLabelLanguageFilter, context: Filters): Flow<String> {
        return codeLogic.listCodeIdsByLabel(filter.region, filter.language, filter.type, filter.label)
    }
}
